packages/snaps-rpc-methods/src/restricted/notify.ts

```diff
@@ -5,31 +5,74 @@ import type {
 } from '@metamask/permission-controller';
 import { PermissionType, SubjectType } from '@metamask/permission-controller';
 import { rpcErrors } from '@metamask/rpc-errors';
-import { NotificationType } from '@metamask/snaps-sdk';
 import type {
   NotifyParams,
   NotifyResult,
-  EnumToUnion,
+  NotificationComponent,
+  InterfaceContext,
 } from '@metamask/snaps-sdk';
-import { type Snap, validateTextLinks } from '@metamask/snaps-utils';
+import {
+  enumValue,
+  NotificationType,
+  union,
+  ContentType,
+  getErrorMessage,
+} from '@metamask/snaps-sdk';
+import { NotificationComponentsStruct } from '@metamask/snaps-sdk/jsx';
+import {
+  createUnion,
+  validateLink,
+  validateTextLinks,
+  type Snap,
+} from '@metamask/snaps-utils';
+import type { InferMatching } from '@metamask/snaps-utils';
+import { object, string } from '@metamask/superstruct';
 import type { NonEmptyArray } from '@metamask/utils';
-import { isObject } from '@metamask/utils';
+import { hasProperty, isObject } from '@metamask/utils';
 
 import { type MethodHooksObject } from '../utils';
 
 const methodName = 'snap_notify';
 
-export type NotificationArgs = {
```

Comment:
This is a breaking change

Reply:
Sorry, I should phrased this as a question, do we need to break this?

Reply:
Yeah, the args have changed?

Reply:
[7dd44f5](https://github.com/MetaMask/snaps/pull/2881/commits/7dd44f509e3028875938d4934d0ca9b8f17bffb9)